tests/app/test_phone_stats_home.py

```diff
@@ -151,3 +152,234 @@ def test_r30_user_must_be_retained_for_at_least_a_month(self):
         # *Now* the user appears in R30.
         r30_results = self.get_success(self.hs.get_datastore().count_r30_users())
         self.assertEqual(r30_results, {"all": 1, "unknown": 1})
+
+
+class PhoneHomeR30V2TestCase(HomeserverTestCase):
+    servlets = [
+        synapse.rest.admin.register_servlets_for_client_rest_resource,
+        room.register_servlets,
+        login.register_servlets,
+    ]
+
+    def _advance_to(self, desired_time: float):
```

Comment:
suggest calling this `desired_time_secs` or something, since it's super unusual for us to use seconds (rather than ms) in Synapse code.